client/lib/feed-stream-store/feed-stream.js

```diff
@@ -488,22 +500,27 @@ export default class FeedStream {
 
 		const postById = this.postById;
 		forEach( this.pendingPostKeys, function( postKey ) {
-			postById.add( postKey.postId );
+			postById.add( keyToString( postKey ) );
 		} );
 
 		const mostRecentPostDate = moment( this.postKeys[ 0 ][ this.dateProperty ] );
 
-		if ( this.pendingDateAfter > mostRecentPostDate ) {
+		// if the first element in the current postKeys isn't also in the pending set,
+		// we have a possible gap
+		const firstKey = this.postKeys[ 0 ];
+		if (
+			! some( this.pendingPostKeys, pendingKey => keysAreEqual( pendingKey, firstKey ) ) &&
+			this.pendingDateAfter > mostRecentPostDate
```

Comment:
2 is the existing condition. If the pendingDateAfter (the last date seen on the set of updates before filtering) doesn't match up with the current mostRecentPostDate, we may have a gap. When we poll for updates, the set we get back is supposed to include the most recent post we already have. If it doesn't, we know we likely have a gap between the current set of updates and the head of the client's stream.